main.py

```diff
@@ -0,0 +1,178 @@
+    def __init__(self):
+        self.database = AccessUserDatabase()
+        self.logged_in_user_id = None
+
+    def hash_password(self, password):
+        """Hashes the password using SHA-256."""
+        hashed_password = hashlib.sha256(password.encode()).hexdigest()
+        return password
+    
+    def verify_username(self, username):
+        good = "good"
+        if username < 1:
+            good = "bad"
+        else:
+            good = "good"
+            if username > 30:
+                good = "bad"
+            else:
+                good = "good"
+        return good
+                
+        
+    
+    def verify_email(self, email):
+        good = "good"
+        if "@" in email:
+            good = "good"
+            if "." not in email:
+                good = "bad"
+            else:
+                good = "good"
+        else:
+            good = "bad"
+        return good
+        
+        
+    def verify_password(self, password, email):
+        good = "good"
+        weak_passwords = ["password", "123456", "qwerty", "abc123"]
+        if len(password) < 8:
+            good = "bad"
+        else:
+            good = "good"
+            if password.lower() in weak_passwords:
+                good = "bad"
+            else:
+                good = "good"
+                if email == password.lower():
+                    good = "bad"
+                else:
+                    good = "good"
+                    has_lower = False
+                    has_upper = False
+                    has_digit = False
+                    has_special = False
+                    for c in password:
+                        if c.islower():
+                            has_lower = True
+                        elif c.isupper():
+                            has_upper = True
+                        elif c.isdigit():
+                            has_digit = True
+                        elif c in "!@#$%^&*()_+":
+                            has_special = True
+                    if has_lower == False:
+                        good = "bad"
+                    if has_upper == False:
+                        good = "bad"
+                    if has_digit == False:
+                        good = "bad"
+                    if has_special == False:
+                        good = "bad"
+        return good
+        
+
+    def sign_up(self):
+        """Allows a new user to sign up."""
+        username = input("Enter a username: ")
+        email = input("Enter an email: ")
+        password = input("Enter a password: ")
+        
+        email = email.lower()
+        
+        
+        if self.verify_username(username) == "bad":
+            print("email not good")
+            return
+        
+        if self.verify_email(email) == "bad":
+            print("email not good")
+            return
+        if self.verify_password(password, email) == "bad":
+            print("email not good")
+            return
```

Comment:
```suggestion
        self.verify_username(username)
        self.verify_email(email)
        self.verify_password(password, email)
```

In my opinion, the print error message and return should be handled in the `verify` functions themselves. The error messages should also be made clearer and with more detail, e.g: "Password doesn't contain a special character."

Reply:
Refactored in [8ac1c5e](https://github.com/rowan04/L5-SE-CW2-QualityPlan/pull/10/commits/8ac1c5e00133d5fbeee4782ca743f2521240516b)